codemodel/model/common/api-version.ts

```diff
@@ -15,7 +15,7 @@
 export interface ApiVersion {
   /** the actual api version string used in the API */
   version: string;
-  range: '-' | '+' | undefined;
+  range?: '-' | '+';
```

Comment:
Perhaps this is a lack of my understanding of TypeScript but what is the difference between these 2? Don't they end up being pretty much the same?

Reply:
They do mean the same thing... almost.

The technical bit is that when constructing an object of this interface, if it's marked `?` then it's optionally supplied (and hence `undefined`) , as opposed to it being mandatory to specify a value (of which `undefined` is a possible choice)

Basically it means that the code has to explicitly say `undefined` vs just accepting that `undefined` is the value you'd get if you didn't specify. 

The compiled code would function exactly the same either way.